### PR TITLE
Add support of file and buffer for the POST Body (#581)

### PR DESCRIPTION
**Pull request by leviliangtw**
1. Make struct Buffer and File independent to Multipart. 
2. Add support of struct Buffer and File to class Body. 
3. Modify CMakeLists.txt.

### CODE REVIEW

File: include/cpr/buffer.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    using data_t = const unsigned char*;
+
+    template <typename Iterator>
+    Buffer(Iterator begin, Iterator end, std::string&& p_filename)
+            // Ignored here since libcurl reqires a long.
+            // There is also no way around the reinterpret_cast.
+            // NOLINTNEXTLINE(google-runtime-int, cppcoreguidelines-pro-type-reinterpret-cast)
+            : data{reinterpret_cast<data_t>(&(*begin))}, datalen{static_cast<long>(std::distance(begin, end))}, filename(std::move(p_filename)) {
+        is_random_access_iterator(begin, end);
+        static_assert(sizeof(*begin) == 1, "only byte buffers can be used");
```
Comment (COM8) [inline include/cpr/buffer.h:18]:
```suggestion
        static_assert(sizeof(*begin) == 1, "Only byte buffers can be used.");
```

**Review by COM8 — CHANGES_REQUESTED** [review]
Looks good. Thanks! Only two minor changes.

**Comment by leviliangtw** [comment]
@COM8 Some tests of CI cannot be passed due to SSL problems, I have no idea what it is. 
However, I think they are not relative to what I modified in my code.

**Review by COM8 — APPROVED** [review]
Looks good now. Thanks.
The Windows/MacOS CI is currently broken. Somehow installing OpenSSL fails. This is something I have to create an issue for.

**Pull request merged**